### ISSUE TITLE
Downstream operation bug when running `sourmash sig cat`

### ISSUE THREAD

**Issue by ccbaumler** [sourmash-bio/sourmash_plugin_directsketch#13]
I have generated a `directsketch` sourmash database from GenBank assembly txt files with:

```
rule gather_sketch_reversioned:
    input:
        reversion = "data/update.{d}-{D}.csv",
    output:
        failed = "data/update.{d}-{D}.failures.csv",
        db = "../dbs/genbank-{d}-{D}.rever.zip"
    conda: "envs/directsketch.yaml"
    resources:
        allowed_jobs = 100
#        mem_mb = 8000,
#        time_min = 30,
    threads: 1
    params:
        k_list = lambda wildcards: ",".join([f"k={k}" for k in KSIZES]),
    shell:'''
        sourmash scripts gbsketch {input.reversion} -o {output.db} --failed {output.failed} \
            --param-str "dna,{params.k_list},scaled=1000,abund" -r 1 #2> {log}
    '''
```

When using the `sourmash sig cat` command to combine newly `directsketch`ed and previous sourmash DBs, the follow error is returned:

 <details>
     <summary>Error Message</summary>

```
$ sourmash sig cat ../dbs/genbank-20240503-fungi.miss.zip ../dbs/genbank-20240503-fungi-k51.clean.zip -k 51 -o ../dbs/genbank-20240503-fungi-k51.zip

== This is sourmash version 4.8.5. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

Traceback (most recent call last):
  File "/home/baumlerc/miniforge3/envs/sourmash/bin/sourmash", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/__main__.py", line 19, in main
    retval = mainmethod(args)
             ^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/cli/sig/cat.py", line 58, in main
    return sourmash.sig.__main__.cat(args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sig/__main__.py", line 130, in cat
    for ss, sigloc in loader:
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sourmash_args.py", line 642, in load_many_signatures
    idx = load_file_as_index(loc, yield_all_files=yield_all_files)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/save_load.py", line 65, in load_file_as_index
    return _load_database(filename, yield_all_files)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/save_load.py", line 113, in _load_database
    db = load_fn(filename,
         ^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/save_load.py", line 190, in _load_sbt
    db = load_sbt_index(filename, cache_size=cache_size)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sbtmh.py", line 10, in load_sbt_index
    return SBT.load(filename, leaf_loader=SigLeaf.load,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sbt.py", line 808, in load
    storage = ZipStorage(location)
              ^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sbt_storage.py", line 107, in __init__
    self._objptr = rustcall(lib.zipstorage_new, to_bytes(path), len(path))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/utils.py", line 78, in rustcall
    raise exc
sourmash.exceptions.Panic: sourmash panicked: thread 'unnamed' panicked with 'called `Result::unwrap()` on an `Err` value: PrependedWithUnknownBytes(4294967296)' at src/core/src/storage.rs:358

```
</details>

This error is also present in other operations like `summarize`

<details>
    <summary>Error Message</summary>

```
sourmash sig summarize ../dbs/genbank-20240503-fungi.miss.zip

== This is sourmash version 4.8.5. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

** loading from '../dbs/genbank-20240503-fungi.miss.zip'
Traceback (most recent call last):
  File "/home/baumlerc/miniforge3/envs/sourmash/bin/sourmash", line 11, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/__main__.py", line 19, in main
    retval = mainmethod(args)
             ^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/cli/sig/fileinfo.py", line 46, in main
    return sourmash.sig.__main__.fileinfo(args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sig/__main__.py", line 1274, in fileinfo
    idx = sourmash_args.load_file_as_index(args.path,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/save_load.py", line 65, in load_file_as_index
    return _load_database(filename, yield_all_files)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/save_load.py", line 113, in _load_database
    db = load_fn(filename,
         ^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/save_load.py", line 190, in _load_sbt
    db = load_sbt_index(filename, cache_size=cache_size)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sbtmh.py", line 10, in load_sbt_index
    return SBT.load(filename, leaf_loader=SigLeaf.load,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sbt.py", line 808, in load
    storage = ZipStorage(location)
              ^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/sbt_storage.py", line 107, in __init__
    self._objptr = rustcall(lib.zipstorage_new, to_bytes(path), len(path))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/baumlerc/miniforge3/envs/sourmash/lib/python3.12/site-packages/sourmash/utils.py", line 78, in rustcall
    raise exc
sourmash.exceptions.Panic: sourmash panicked: thread 'unnamed' panicked with 'called `Result::unwrap()` on an `Err` value: PrependedWithUnknownBytes(4294967296)' at src/core/src/storage.rs:358
```

</details>

This bug is present in the latest version ran on 04 May 2024 (i.e. version 0.1.0).

**Comment by bluegenes** [comment]
v0.2.0 should solve this problem -- can you test and then let me know? thx!

**Comment by bluegenes** [comment]
closing, as the issue was solved (and we're now on v0.3.x)
